content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -463,20 +465,20 @@ Encapsulation wraps both code and data together
 
 Abstraction hides complexity while encapsulation hides the internal working
 
-# Conclusion
+### Conclusion
 
-To this point, we’ve demystified the concept of OOP. To say the least, abstraction and encapsulation are two sides of the same coin. I hope you find this insightful. I hope to write about Inheritance and Polymorphism in my next OOP concept article.
+We have successfully learned about the concept of OOP starting with abstraction and encapsulation. You have learnt how to declare a class and create and object. You have also learnt how to implement abstraction and encapsulation,and you have seen also the differences between abstraction and encapsulation.
 
 ### References
 
 Java- How to program 10th Ed-Early Object Version- Deitel
 
 Introduction to Java Programming (10th ed) Comprehensive Version [Liang- 2014-01-06]
 
-[oracle java docs](https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html)
+[oracle java docs abstract methods and classes](https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html)
 
-[oracle java docs](https://docs.oracle.com/javase/tutorial/java/concepts/index.html)
+[oracle java docs OOP concept](https://docs.oracle.com/javase/tutorial/java/concepts/index.html)
 
-[w3schools]([https://www.w3schools.com/java/java_abstract.asp)
+[w3schools abstract class]([https://www.w3schools.com/java/java_abstract.asp)
```

Comment:
> \[w3schools abstract class\]\(\[https://www.w3schools.com/java/java_abstract.asp)

This link doesn't work, do this to fix it:

> \[w3schools abstract class\]\(https://www.w3schools.com/java/java_abstract.asp)

Notice how I removed the extra *[* between the brackets.